### PR TITLE
Replace Maint. Drone machinery whitelist with last touched check & other tweaks

### PR DESCRIPTION
**Pull request by JJRcop**
## About The Pull Request

@ninjanomnom came up with the idea to replace the machine whitelist on drones with a restriction against machines used in the last 5 minutes by another being. I asked the headmins what they thought of the idea, and they gave their approval.

Air Alarms and most Computers are still blocked due to being high risk abuse vectors, but the Chief Engineer can enable them on his PDA. This also has headmin approval.

I've also done a little tidying up around some of the other rough edges with drones

 * Fixed drone minds being named "drone" instead of "Drone (563)"
     * This fix applies to any corpse with unique name enabled
 * Drones can use chisels and wallframes
 * Simple robots like the cleanbot and floorbot will not restrict drones anymore
 * Drones now respect the USE_EXP_RESTRICTIONS_ADMIN_BYPASS config var

Here are some additional changes in the pull request not related to drones

* `/atom/proc/tool_act` returns a bitflag instead of TRUE
* Air alarm now properly uses `tooltype_act` instead of the legacy `attackby` override

## Why It's Good For The Game
Opens up many more interactions for drones when it comes to constructing and maintaining different machines. 

## Changelog
:cl: JJRcop, ninjanomnom
qol: Instead of a machinery whitelist, Drones cannot interact with machines which were touched by a being in the past 5 minutes.
qol: Drones have a new machinery blacklist which currently contains the Air Alarm, and most Computers. The CE can disable this on their PDA.
qol: Drones aren't blocked by simple robots anymore. (Cleanbot, Floorbot, etc)
qol: Drones can now use chisels, and construct APCs, newscasters, lights, and other things placed on walls.
fix: Drones now respect USE_EXP_RESTRICTIONS_ADMIN_BYPASS
/:cl:

### CODE REVIEW

File: code/datums/wires/explosive.dm [hunk 1]
```diff
@@ -66,6 +68,8 @@
 	holder_type = /obj/item/grenade/c4
 
 /datum/wires/explosive/c4/interactable(mob/user) // No need to unscrew wire panels on plastic explosives
+	if(!..())
+		return TRUE
 	return TRUE
```
Comment (ATH1909) [inline code/datums/wires/explosive.dm:73]: ????

File: code/modules/mob/living/simple_animal/friendly/drone/_drone.dm [hunk 2]
```diff
@@ -366,19 +354,19 @@
 
 /mob/living/simple_animal/drone/proc/shy_update()
 	var/list/drone_bad_areas = make_associative(drone_area_blacklist_flat) + typecacheof(drone_area_blacklist_recursive)
-	var/list/drone_good_machines = make_associative(drone_machinery_whitelist_flat) + typecacheof(drone_machinery_whitelist_recursive)
 	var/list/drone_good_items = make_associative(drone_item_whitelist_flat) + typecacheof(drone_item_whitelist_recursive)
+	var/static/list/not_shy_of = typecacheof(list(/mob/living/simple_animal/drone, /mob/living/simple_animal/bot))
```
Comment (ATH1909) [inline code/modules/mob/living/simple_animal/friendly/drone/_drone.dm:358]: does this mean that drones aren't shy of pAI-controlled bots or of wizards who have shapeshifted into ED-209s?

Comment (JJRcop) [reply code/modules/mob/living/simple_animal/friendly/drone/_drone.dm:358]: that's right. i think it's too much of an edge case to worry about

File: code/datums/wires/explosive.dm (hunk 1 above)
Comment (JJRcop) [reply code/datums/wires/explosive.dm:73]: Haha, I did find and replace, so this happened. The parent proc returns true anyway, so this definition is useless

Comment (JJRcop) [reply code/datums/wires/explosive.dm:73]: I don't even know man... What exactly was I thinking?

**Comment by JJRcop** [comment]
Received headmin approval for new machinery blacklist, with CE gaining the ability to disable this.

**Comment by JJRcop** [comment]
Proposal: Add exosuit fabricator to blacklist so drones can't print more drones. Thoughts?

**Comment by JJRcop** [comment]
> Proposal: Add exosuit fabricator to blacklist so drones can't print more drones. Thoughts?
I've decided no, but I did add computers to the blacklist

**Comment by JJRcop** [comment]
Coming up before undraft: drones can use certain computers
ETA: later this afternoon

**Comment by JJRcop** [comment]
Whitelisted arcades, and some computers.

**Review by Mothblocks — CHANGES_REQUESTED** [review]

File: code/datums/components/technointrovert.dm [hunk 3]
```diff
@@ -29,8 +30,19 @@
...
+	SIGNAL_HANDLER
+	if(!is_in_whitelist(source, machine))
+		return COMPONENT_CANT_USE_MACHINE_INTERACT | COMPONENT_CANT_USE_MACHINE_TOOLS
 
+/datum/component/technointrovert/proc/on_try_wires_interact(datum/source, atom/machine)
+	SIGNAL_HANDLER
+	if(!ismachinery(machine))
+		return
+	else if(!is_in_whitelist(source, machine))
+		return COMPONENT_CANT_INTERACT_WIRES
```
[the remark this answers is not in the document]
Comment (JJRcop) [reply code/datums/components/technointrovert.dm:48]: I feel pretty silly now

File: code/_onclick/item_attack.dm [hunk 4]
```diff
@@ -11,7 +11,7 @@
 
 	var/is_right_clicking = LAZYACCESS(params2list(params), RIGHT_CLICK)
 
-	if(tool_behaviour && target.tool_act(user, src, tool_behaviour, is_right_clicking))
+	if(tool_behaviour && (target.tool_act(user, src, tool_behaviour, is_right_clicking) & TOOL_ACT_MELEE_CHAIN_BLOCKING))
```
[the remark this answers is not in the document]
Comment (JJRcop) [reply code/_onclick/item_attack.dm:14]: This is the only consumer

**Comment by JJRcop** [comment]
Requested changes addressed.

**Pull request merged**